### ISSUE TITLE
Revamping the Mattermost community FAQ

### ISSUE THREAD

**Issue by azigler** [mattermost/docs#6851]
The [Mattermost community FAQ](https://docs.mattermost.com/about/faq-community.html) is outdated and has limited content. It would benefit from a refresh for 2024, pulling from our experiences working with the community and answering their questions. Right now it has only 2 Q&As:

- How can I contribute to Mattermost?
- Can contributors add themselves to the Mattermost company page on LinkedIn?

The first question ("how") needs to be revised to point to ["Why and how to contribute"](https://developers.mattermost.com/contribute/why-contribute/) so we're not duplicating work. And this Q&A doesn't offer much of an engagement hook currently.

The second question ("can") needs to be reworked -- does anyone even do this? I argue that the best way for contributors to show their Mattermost pride is with badges and certification, which is on my DevRel objective list for 2024.

I think this page needs several more Q&As to justify it's existence. Otherwise, it risks being hidden, duplicative information. Here are some potential questions to kick us off:

- What is Mattermost and how can I contribute to its open-source projects?
- Do I need to be skilled in a specific programming language to contribute to Mattermost?
- How do I set up my development environment for Mattermost?
- Can I contribute to Mattermost if I don't have a lot of experience in open-source projects?
- How does Mattermost manage contributions and what is the process for getting my contributions accepted?
- How can I stay updated with the latest developments and discussions in the Mattermost community?
- What is the process for proposing a new feature or significant change to Mattermost?
- Can I work on a bug or feature that's not listed as a 'Good First Issue'? How do I proceed?
- How can I find a mentor or get more personalized guidance as I start contributing?
- What are some common mistakes or pitfalls new contributors should avoid when contributing to Mattermost?

What do we think of those questions? Do we need all of them? Are there better substitutions or additions? Tagging some community-minded staff to kick off discussion. @cwarnermm @larkox @hmhealey @lindy65 @mickmister @spirosoik @M-ZubairAhmed @jasonblais @jespino @hanzei

**Comment by jasonblais** [comment]
Thanks @azigler! Excellent questions to start with. Do we know what areas our community can use more guidance / help around?

We also have some content in our handbook which can help:
https://handbook.mattermost.com/contributors/contributors
https://handbook.mattermost.com/contributors/contributors/ways-to-contribute
https://handbook.mattermost.com/contributors/contributors/community

**Comment by lindy65** [comment]
Thanks @azigler - agree with Jason that these are all great questions to start with. 

I'd be happy to work with you to include / improve QA community contributor information

**Comment by larkox** [comment]
@azigler Here are a few more. Some of them are just rephrasing some of your questions in a way I like them more. But leave up to you what is the best :stuck_out_tongue_winking_eye: 

- I want to contribute to Mattermost. Where do I start? (The idea here is to tell them to go to the issues and look for the up for grabs label, and explain what the other labels mean)
- I submitted my PR. Now what? (Explain the review process: potential UX and Product reviews, 2 Dev reviews and 1 QA review)
- I don't know how to code. Can I contribute to Mattermost? (Explain docs, translations and QA contributions programs, and not sure if we have any other)
- I'm stuck on an issue. What do I do? (Explain how to interact with the staff in the community server)
- Can I take more than one issue at a time? (This is something I have been asked a few times. The general answer is yes, but don't hoard, and don't take more than what you can handle)
- I got really busy and cannot work on an issue I have assigned. What do I do? (Close the PR if it is open, and post a message in the issues list. Maybe we could add that help wanted issues are not time sensitive, and that they can take their time, but if they are sure they won't work on them, better close them and let someone else take the mantle)
- My PR has been sitting for a while without anyone assigned. What should I do? (Remind them that we are human, and if more than X time passes, drop a line in the community server so someone looks at it)
- I addressed the changes in my PR but I received no answer from the reviewers. What should I do? (Remind them they can push the "re-request review" button, if they have done that, remind them we are human, and if more than X time has passed, to feel free to leave a comment on the PR mentioning the reviewer)

If I think of any other question I will post it here.

**Comment by M-ZubairAhmed** [comment]
@azigler a little more from my side

1. There is an issue assigned to a contributor, but it hasn't been addressed for some time. I'd like to take on the task. How can I proceed?
1. How can I run CI pipelines locally to avoid waiting for the PR CI?
1. I'm a bit hesitant to post a question in the open channel. How can I identify the person responsible for the open issue? In other words, who is the owner of the issue and can provide more context?
1. I started contributing, but now I'm stuck in the middle. How to share code (Open a draft PR)
1. I'm actively working on an issue, but it requires a license to unlock. How can I obtain the necessary license?

Troubleshooting the setup of a local server seems to be a significant hurdle for first-time contributors. I suggest creating a dedicated FAQ on this topic, compiling common troubleshooting steps. It could also be linked to existing conversations within the community. I'm open to suggestions and ideas.
